app/routes.py

```diff
@@ -1 +1,224 @@
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return {"goal": goal.to_dict()}, 200    
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    title = goal.title
+    db.session.delete(goal)
+    db.session.commit()
+
+    return {"details": f'Goal {goal_id} "{title}" successfully deleted'}, 200
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    if "completed_at" in request_body and ((isinstance(request_body["completed_at"], datetime) or request_body["completed_at"]) == None):
+        new_task = Task.from_dict(request_body)
+    elif "completed_at" in request_body and not isinstance(request_body["completed_at"], datetime) and not request_body["completed_at"] == None:
+        return make_response({"details": "Completed_at must be a datetime"}, 400)
+    else:
+        new_task = Task(
+            title = request_body["title"],
+            description = request_body["description"]
+        )
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task":new_task.to_dict()}, 201
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    alpha_sort_query = request.args.get("sort")
+    id_query = request.args.get("id")
+    title_query = request.args.get("title")
+
+    if alpha_sort_query is None:
+        all_tasks = Task.query.all()
+    elif alpha_sort_query == "asc":
+        all_tasks = Task.query.order_by(Task.title.asc())
+    elif alpha_sort_query == "desc":
+        all_tasks = Task.query.order_by(Task.title.desc())
+    
+    if alpha_sort_query is None and id_query is None and title_query is None:
+        all_tasks = Task.query.all()
+    elif alpha_sort_query == "asc":
+        if id_query is None and title_query is None:
+            all_tasks = Task.query.order_by(Task.title.asc())
+        elif id_query == "asc":
+            all_tasks = Task.query.order_by(Task.title.asc(), Task.task_id.asc())
+        elif id_query == "desc":
+            all_tasks = Task.query.order_by(Task.title.asc(), Task.task_id.desc())
+    elif alpha_sort_query == "desc":
+        if id_query is None and title_query is None:
+            all_tasks = Task.query.order_by(Task.title.desc())
+        elif id_query == "asc":
+            all_tasks = Task.query.order_by(Task.title.desc(), Task.task_id.asc())
+        elif id_query == "desc":
+            all_tasks = Task.query.order_by(Task.title.desc(), Task.task_id.desc())
+    elif id_query == "asc":
+        all_tasks = Task.query.order_by(Task.task_id.asc())
+    elif id_query == "desc":
+        all_tasks = Task.query.order_by(Task.task_id.desc())
+    elif title_query is not None:
+        all_tasks = Task.query.filter_by(title=title_query)
```

Comment:
1. I am incredibly impressed that you managed to write such a long chain of conditionals. It's so big I kinda love it. Wow. 🤩 
2. Even though I love it, it is bad practice to have this many conditional statements because it negatively impacts readability/maintainability. So unfortunately I do have to recommend refactoring this down.